CrossrefReferenceLinkingPlugin.php

```diff
@@ -0,0 +1,474 @@
+	{
+		$response = & $params[1];
+		$submission = & $params[2];
+		// Get DOMDocument from the response XML string
+		$xmlDoc = new DOMDocument();
+		$xmlDoc->loadXML($response);
+		if ($xmlDoc->getElementsByTagName('citations_diagnostic')->length > 0) {
+			$citationsDiagnosticNode = $xmlDoc->getElementsByTagName('citations_diagnostic')->item(0); /** @var DOMNodeList $citationsDiagnosticNode */
+			$citationsDiagnosticCode = $citationsDiagnosticNode->getAttribute('deferred') ;
+			//set the citations diagnostic code and the setting for the automatic check
+			$submission->setData($this->getCitationsDiagnosticIdSettingName(), $citationsDiagnosticCode);
+			$submission->setData($this->getAutoCheckSettingName(), true);
+			$submission = Repo::submission()->edit($submission, array(), Application::get()->getRequest());
+		}
+		return false;
+	}
+
+	/**
+	 * Add properties to the submission entity (SchemaDAO-based)
+	 *
+	 * @param $hookName string `Schema::get::submission`
+     * @param array $args [
+     *      @option stdClass $schema
+     * ]
+	 */
+	public function addSubmissionSchema(string $hookName, array $args): bool
+	{
+		$schema = $args[0];
+
+		$schema->properties->{$this->getCitationsDiagnosticIdSettingName()} = (object) [
+			'type' => 'string',
+			'apiSummary' => true,
+			'validation' => ['nullable']
+		];
+
+		$schema->properties->{$this->getAutoCheckSettingName()} = (object) [
+			'type' => 'boolean',
+			'apiSummary' => true,
+			'validation' => ['nullable']
+		];
+		return false;
+	}
+
+	/**
+	 * Consider the additional citation setting name 'crossref::doi'.
+	 */
+	public function getAdditionalCitationFieldNames(string $hookName, CitationDAO $citationDao, array &$additionalFields): bool
+	{
+		$additionalFields[] = $this->getCitationDoiSettingName();
+		return false;
+	}
+
+	/**
+	 * Resets the submission data related to Reference Linking Plugin.
+	 * Used every time the citations for a certain publication are imported.
+	 *
+	 * @param $hookName string 'CitationDAO::afterImportCitations'
+	 * @param $params array [
+	 *  @option integer The publication ID for which the citations are imported
+	 * ]
+	 */
+	public function citationsChanged(string $hookName, array $params): bool
+	{
+		$publicationId = $params[0];
+
+		$publication = Repo::publication()->get($publicationId);
+		$submission = Repo::submission()->get($publication->getData('submissionId'));
+
+		if ($submission->getData($this->getCitationsDiagnosticIdSettingName())) {
+			$submission->setData($this->getCitationsDiagnosticIdSettingName(), null);
+			$submission->setData($this->getAutoCheckSettingName(), null);
+			$submission = Repo::submission()->edit($submission, array(), Application::get()->getRequest());
+		}
+		return false;
+	}
+
+	/**
+	 * Get found Crossref references DOIs for the given publication DOI.
+	 */
+	public function getCrossrefReferencesDOIs(Publication $publication): void
```

Comment:
I think the name needs to be improved, as it's prefixed with `get` but doesn't retrieve anything.